cmake/Modules/BuildCommon.cmake

```diff
@@ -484,7 +482,7 @@ function(add_katana_sphinx_target target_name)
       COMMAND ${CMAKE_COMMAND} -E echo "${target_name} documentation at file://${CMAKE_BINARY_DIR}/docs/${target_name}/index.html"
       COMMENT "Building internal ${target_name} sphinx documentation"
     )
-  else ()
+  elseif (BUILD_DOCS STREQUAL "external")
     add_custom_target(
```

Comment:
can we dedupe this `add_custom_target` with the snippet I gave?

Reply:
No. Doesn't get picked up by sphinx-build.

Reply:
If we are going to write 
```cmake
if()
  add_custom_target(... a)
else()
  add_custom_target(... b)
endif()
```

instead of 

```cmake
set(var b)
if ()
  set(var a)
endif()
add_custom_target(... ${var})
```

I'm going to need a little more explanation than "it doesn't get picked up by sphinx-build".

Reply:
Even the following doesn't work. Should I add a comment to make sure it doesn't get reverted?
```
set(targ_options "-t external")
add_custom_target(... sphinx-build ${tag-options})
```